src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -556,16 +573,26 @@ private static CompletableFuture<?> checkSSLConfig(AbstractAsyncSSLConnection ac
                 .thenCompose(checkAlpnCF);
     }
 
-    synchronized boolean finalStream() {
-        return finalStream;
+    boolean finalStream() {
+        stateLock.lock();
+        try {
+            return finalStream;
```

Comment:
Perhaps we can make `finalStream` volatile and remove the use of locks when accessing and setting it?

Reply:
Looks doable